apis/keda/v1alpha1/triggerauthentication_types.go

```diff
@@ -231,3 +247,44 @@ func init() {
 	SchemeBuilder.Register(&ClusterTriggerAuthentication{}, &ClusterTriggerAuthenticationList{})
 	SchemeBuilder.Register(&TriggerAuthentication{}, &TriggerAuthenticationList{})
 }
+
+// interface that can handle TriggerAuthentication/ClusterTriggerAuthentication generically
+type TriggerAuthenticationStatusHandler interface {
+	metav1.Object
+	GetScaledObjectNameStr() string
+	SetScaledObjectNameStr(str string)
+	GetScaledJobNameStr() string
+	SetScaledJobNameStr(str string)
+}
+
+func (t *TriggerAuthentication) GetScaledObjectNameStr() string {
+	return t.Status.ScaledObjectNamesStr
+}
+
+func (t *TriggerAuthentication) SetScaledObjectNameStr(str string) {
+	t.Status.ScaledObjectNamesStr = str
+}
+
+func (t *TriggerAuthentication) GetScaledJobNameStr() string {
+	return t.Status.ScaledJobNamesStr
+}
+
+func (t *TriggerAuthentication) SetScaledJobNameStr(str string) {
+	t.Status.ScaledJobNamesStr = str
+}
+
+func (c *ClusterTriggerAuthentication) GetScaledObjectNameStr() string {
+	return c.Status.ScaledObjectNamesStr
+}
+
+func (c *ClusterTriggerAuthentication) SetScaledObjectNameStr(str string) {
+	c.Status.ScaledObjectNamesStr = str
+}
+
+func (c *ClusterTriggerAuthentication) GetScaledJobNameStr() string {
+	return c.Status.ScaledJobNamesStr
+}
+
+func (c *ClusterTriggerAuthentication) SetScaledJobNameStr(str string) {
```

Comment:
I am not sure this is the right approach. Wouldn't be better to use methods on `TriggerAuthenticationStatus` which do everything in one step? Something like `AddScaledObjectName(soName string)`/`RemoveScaledObjectName(soName string)`. It would do all the magic inside (`kedacontrollerutil.AppendIntoString`...). These methods could probably return just a new Status that should be then applied to the cluster (please use `client.Patch()` isntead of  `Update()`). You can check how do we update Status for ScaleObjects: https://github.com/kedacore/keda/blob/main/pkg/util/status.go